File: src/main/java/Problem0035.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashSet;

public class Problem0035 {

    private final ArrayList<Integer> primes;

    public Problem0035() {
        long startTime = System.nanoTime();
        long result = 0;

        // Solution:
        this.primes = this.genPrimes(1000000);
        HashSet<Integer> circularPrimes = new HashSet<>();
        for (Integer prime : this.primes) {
            if (circularPrimes.contains(prime)) continue;
            Integer[] primeSwapped = this.swapped(prime);

            if (primeSwapped != null) {
                circularPrimes.addAll(Arrays.asList(primeSwapped));
            }
        }
        result = circularPrimes.size();

        long timeToResolve = System.nanoTime() - startTime;
        System.out.println("Result:\t" + result + "\tTime:\t" + (((double) timeToResolve / 1_000_000) > 1000 ?
                (((double) timeToResolve / 1_000_000_000) + "s") :
                (((double) timeToResolve / 1_000_000) + "ms")));
    }

    private Integer[] swapped(int n) {
        String asString = n + "";

        //Optimized:
        if (n > 10 && (asString.contains("0") || asString.contains("2") || asString.contains("4") || asString.contains("5") || asString.contains("6") || asString.contains("8"))) {
            return null;
        }

        String[] permutations = new String[asString.length()];
        HashSet<Integer> primeSwapped = new HashSet<>();
        permutations[0] = asString;
        primeSwapped.add(Integer.parseInt(asString));

        for (int i = 1; i < permutations.length; i++) {
            permutations[i] = (permutations[i - 1].substring(1) + permutations[i - 1].charAt(0));
            if (!this.primes.contains(Integer.parseInt(permutations[i]))) {
                return null;
            }
            primeSwapped.add(Integer.parseInt(permutations[i]));
        }
        return primeSwapped.toArray(new Integer[0]);
    }

    public ArrayList<Integer> genPrimes(int range) {
        ArrayList<Integer> primes = new ArrayList<>();
        primes.add(2);
        outer:
        for (int i = 3; i < range; i += 2) {
            int sqrt = (int) Math.sqrt(i);
            for (int j = 0; j < primes.size() && primes.get(j) <= sqrt; j++) {
                if (i % primes.get(j) == 0) {
                    continue outer;
                }
            }
            primes.add(i);
        }
        return primes;
    }
}

File: src/main/cs/Problem0035.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using DocumentFormat.OpenXml.Drawing.Charts;
using Euler.test.cs;

namespace Euler.main.cs
{
    public class Problem0035
    {
        private List<int> primes;

        public Problem0035()
        {
            Stopwatch stopWatch = new Stopwatch();
            stopWatch.Start();
            long result = 0;

            // Solution:
            primes = genPrimes(1000000);
            HashSet<int> circularPrimes = new HashSet<int>();
            foreach (int prime in primes)
            {
                if (circularPrimes.Contains(prime)) continue;
                int[] primeSwapped = swapped(prime);

                if (primeSwapped != null)
                {
                    foreach (int i in primeSwapped)
                    {
                        circularPrimes.Add(i);
                    }
                }
            }

            result = circularPrimes.Count;


            stopWatch.Stop();
            string elapsedTime = stopWatch.Elapsed.ToString();
            Console.WriteLine("Result:\t" + result + "\tTime:\t" +
                              (double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1, 2)) >= 1
                                  ? double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1)) + " s"
                                  : double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000 +
                                    " ms"));
            if (Test.DoBenchmark)
            {
                Benchmark.AddTime(35, double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000);
            }
        }

        private int[] swapped(int n)
        {
            String asString = n + "";

            //Optimized:
            if (n > 10 && (asString.Contains("0") || asString.Contains("2") || asString.Contains("4") ||
                           asString.Contains("5") || asString.Contains("6") || asString.Contains("8")))
            {
                return null;
            }

            String[] permutations = new String[asString.Length];
            HashSet<int> primeSwapped = new HashSet<int>();
            permutations[0] = asString;
            primeSwapped.Add(int.Parse(asString));

            for (int i = 1; i < permutations.Length; i++)
            {
                permutations[i] = (permutations[i - 1].Substring(1) + permutations[i - 1][0]);
                if (!primes.Contains(int.Parse(permutations[i])))
                {
                    return null;
                }

                primeSwapped.Add(int.Parse(permutations[i]));
            }

            return primeSwapped.ToArray();
        }

        private List<int> genPrimes(int range)
        {
            List<int> primes = new List<int> {2};
            for (int i = 3; i < range; i += 2)
            {
                int sqrt = (int) Math.Sqrt(i);
                for (int j = 0; j < primes.Count && primes[j] <= sqrt; j++)
                {
                    if (i % primes[j] == 0) goto skip;
                }

                primes.Add(i);
                skip:
                {
                }
            }

            return primes;
        }
    }
}